partition_table.go

```diff
@@ -194,43 +194,58 @@ func (p *PartitionTable) Close() error {
 func (p *PartitionTable) createStorage(ctx context.Context) (*storageProxy, error) {
 	var (
 		err  error
+		errs = new(multierr.Errors)
 		st   storage.Storage
 		done = make(chan struct{})
 	)
+
 	start := time.Now()
 	ticker := time.NewTicker(5 * time.Minute)
 	defer ticker.Stop()
+	defer close(done)
 	go func() {
-		defer close(done)
-		st, err = p.builder(p.topic, p.partition)
-		if err != nil {
-			return
+		for {
+			select {
+			case <-ctx.Done():
```

Comment:
Do you need to wait for the context here also?
I guess just stopping the logging via the `done` channel is fine.
The context is then only used outside of the logging goroutine to track the storage creation

Reply:
I would also remove the `case <-ctx.Done():` because if the context is done, we will not stop the storage building. So the logs should also not stop.

Reply:
Yup, makes sense.